scipy/stats/_wilcoxon.py

```diff
@@ -222,8 +213,26 @@ def _wilcoxon_nd(x, y=None, zero_method='wilcox', correction=True,
     # the decision was to use an exact test, we override this to
     # a permutation test now (since method=`exact`` is not exact in the
     # presence of ties)
-    if m_auto and has_ties and method == 'exact':
-        method = stats.PermutationMethod()
+    if method == "auto":
+        if not (has_ties or has_zeros):
+            method = "exact"
+        else:
+            # the possible outcomes to be simulated by the permutation test
+            # are 2**n, where n is the sample size.
+            if d.shape[-1] > 13:
+                n_resamples = 9999
+                msg = ("Since the sample size is not large and there are ties "
+                       "or zeros, method=`auto` performs a permutation test. "
+                       "For sample size > 13, the p-value is simulated and "
+                       "not deterministic since `n_resamples` is capped at "
+                       "9999 to avoid long runtimes. Consider running "
+                       "wilcoxon using PermutationMethod with a larger "
```

Comment:
```suggestion
                       "`wilcoxon` using a `PermutationMethod` with a larger "
```
and I think the first two sentences can be combined for concision, e.g.
> ... method=`auto` performs a randomized permutation test with 9999 resamples to limit execution time. Consider running...